tests/neg/applied_constructor_types.scala

```diff
@@ -0,0 +1,10 @@
+import scala.language.experimental.modularity
+
+def f(x: Int): Int = x
+def id[T](x: T): T = x
+def idDependent(x: Any): x.type = x
+
```

Comment:
What about constructors with mixed tracked and non-tracked parameters?

Reply:
Added some tests for those, but the current approach is: you have to write all parameters, but the non-tracked params won't influence the resulting type.

Reply:
that sounds strange, so a bit like an unused function parameter?

Reply:
It doesn't have to be unused, though one might argue that it makes it worse :sweat_smile: BUT, the following works:
```scala
class Generic[A](val a: A)
val _: Generic(compiletime.erasedValue[Int]) = Generic(42)
val _: Generic(??? : Int) = Generic(42)
val _: Generic(43) = Generic(42)
```